File: Q1.-Set-Mismatch.cpp
1class Solution {
2public:
3    vector<int> findErrorNums(vector<int>& nums) {
4        int n = nums.size();
5        int expected_sum = n * (n + 1) / 2;
6        int arr_sum = accumulate(nums.begin(), nums.end(), 0);
7        set<int> res(nums.begin(), nums.end());
8        int unique_sum = accumulate(res.begin(), res.end(), 0);
9        int ele = expected_sum - unique_sum;
10        int dup = arr_sum - unique_sum;
11        return {dup, ele};
12    }
13};

File: Q1.-Set-Mismatch.py
1class Solution:
2    def findErrorNums(self, nums: List[int]) -> List[int]:
3        arr=[]
4        counts=Counter(nums)
5        res=set(nums)
6        n=len(nums)
7        for i in range(1,n+1):
8            if counts[i]>1:
9                dup=i
10            if i not in res:
11                miss=i
12
13        return [dup,miss]        
14
15
16
17    
18
19            